projects/noloan/admin/app/src/main/java/hasadna/noloan/admin/app/DBMessagesHolder.java

```diff
@@ -0,0 +1,103 @@
+package hasadna.noloan.admin.app;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import hasadna.noloan.protobuf.SmsProto.SmsMessage;
+
+// Simple singleton to hold the suggestions list
+public class DBMessagesHolder {
+  private static DBMessagesHolder instance;
+  private List<SmsMessage> spam;
+  private List<SmsMessage> suggestions;
+
+  MessagesListener spamListener;
+  MessagesListener suggestionsListener;
+
+  public static DBMessagesHolder getInstance() {
+    if (instance == null) {
+      instance = new DBMessagesHolder();
+    }
+    return instance;
+  }
+
+  public DBMessagesHolder() {
+    spam = new ArrayList<>();
+    suggestions = new ArrayList<>();
+  }
+
+  public void init(List<SmsMessage> spam, List<SmsMessage> suggestions) {
+    this.spam = spam;
+    this.suggestions = suggestions;
+  }
+
+  public List<SmsMessage> getSpam() {
+    return spam;
+  }
+
+  public List<SmsMessage> getSuggestions() {
+    return suggestions;
+  }
+
+  public void addSpam(SmsMessage spam) {
+    this.spam.add(spam);
+
+    if (spamListener != null) {
+      spamListener.messageAdded();
+    }
+  }
+
+  public void spamModified(SmsMessage spam) {
```

Comment:
If `spam` was changed, you won't be able to find it in `spam.indexOf()`.

Reply:
Tom fixed it in his PR

Reply:
Ok